### DISCUSSION TITLE
How to get a pfm file with little endian structure?

### DISCUSSION THREAD

**Discussion by NotADinosaur** [ImageMagick/ImageMagick#3744] [Help] [answered]
when running `magick convert blank.png blank.pfm` it gives me a .pfm file with the big endian structure. Is there an extra tag I could add to make it use the little endian structure?

**Comment by fmw42** [comment, marked as answer]
See -endian at https://imagemagick.org/script/command-line-options.php#endian

Note in ImageMagick 7, use magick, not magick convert.

**Reply by NotADinosaur** [reply]
Would `magick blank.png blank2.pfm -endian LSB` be right? I don’t think it is because it returns `magick: unable to open image blank2.pfm: no such file or directory @ error/blob.c/openblob/3537`

**Reply by fmw42** [reply]
`magick black.png -endian LSB black2.pfm`

assuming PFM allows LSB

**Reply by snibgo** [reply]
@NotADinosaur: you suggested:
```
magick blank.png blank2.pfm -endian LSB
```
No, that is wrong. You should read the input, then set the endian to modify the behaviour of the writing, then write the output, like this:
```
magick blank.png -endian LSB blank2.pfm
```

**Reply by NotADinosaur** [reply]
Thank you! Sorry, I’m not good with command-based stuff.
